clang/lib/Parse/ParseExpr.cpp

```diff
@@ -2422,6 +2425,39 @@ ExprResult Parser::ParseSYCLUniqueStableNameExpression() {
                                                T.getCloseLocation(), Ty.get());
 }
 
+// Parse a __builtin_sycl_unique_stable_id expression. Accepts an expression,
+// but we later ensure that it MUST be a DeclRefExpr to a VarDecl of some form.
+ExprResult Parser::ParseSYCLUniqueStableIdExpression() {
+  assert(Tok.is(tok::kw___builtin_sycl_unique_stable_id) &&
+         "Not __bulitin_sycl_unique_stable_id");
+
+  SourceLocation OpLoc = ConsumeToken();
+  BalancedDelimiterTracker T(*this, tok::l_paren);
+
+  if (T.expectAndConsume(diag::err_expected_lparen_after,
+                         "__builtin_sycl_unique_stable_id"))
+    return ExprError();
+
+  EnterExpressionEvaluationContext ConstantEvaluated(
+      Actions, Sema::ExpressionEvaluationContext::ConstantEvaluated);
+
+  ExprResult VarExpr =
+      Actions.CorrectDelayedTyposInExpr(ParseAssignmentExpression());
+
+  if (!VarExpr.isUsable()) {
+    T.skipToEnd();
+    return ExprError();
+  }
+
+  if (T.consumeClose())
+    return ExprError();
+
+  // BuildSYCLUniqueStableIdExpr will do the Expr checking to make sure we are
```

Comment:
Comment is a bit twisty -- we eventually get there, but we call the Act function here. I'd probably just drop the comment as it doesn't add a whole lot; everyone knows Sema is where the semantic checking happens anyway.